apix-impl/src/main/java/eu/xenit/apix/alfresco/sites/SiteService.java

```diff
@@ -0,0 +1,73 @@
+package eu.xenit.apix.alfresco.sites;
+
+import com.github.dynamicextensionsalfresco.osgi.OsgiService;
+import eu.xenit.apix.alfresco.ApixToAlfrescoConversion;
+import eu.xenit.apix.data.NodeRef;
+import eu.xenit.apix.sites.ISite;
+import eu.xenit.apix.sites.ISiteService;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.alfresco.service.ServiceRegistry;
+import org.alfresco.service.cmr.site.SiteInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@OsgiService
+@Component("eu.xenit.apix.alfresco.sites.SiteService")
+public class SiteService implements ISiteService {
+
+    private final static Logger logger = LoggerFactory.getLogger(SiteService.class);
+    private final static String DOCUMENT_LIBRARY_COMPONENT = "documentLibrary";
+    private final static String LINKS_COMPONENT = "links";
+    private final static String DATA_LISTS_COMPONENT = "dataLists";
+    private final static String WIKI_COMPONENT = "wiki";
+    private final static String DISCUSSIONS_COMPONENT = "discussions";
+
+    private ApixToAlfrescoConversion c;
+
+    @Autowired
+    private ServiceRegistry serviceRegistry;
```

Comment:
Both field **and** constructor injection are being used here. I don't think that is right.
I know APIX is a bit of a mess when it comes to injection, but since this is a new class, let's use 100% constructor injection.